custom/icds/management/commands/get_icds_sms_usage.py

```diff
@@ -119,25 +137,62 @@ def handle(self, domain, start_date, end_date, **options):
         ):
             location = self.get_location(sms)
             state_code = self.get_state_code(location)
-            if state_code not in data:
-                data[state_code] = {}
+            district_code = self.get_district_code(location)
+            if state_code not in district_level_data:
+                district_level_data[state_code] = {}
+            if district_code not in district_level_data[state_code]:
+                district_level_data[state_code][district_code] = {}
 
             indicator_slug = self.get_indicator_slug(sms)
-            if indicator_slug not in data[state_code]:
-                data[state_code][indicator_slug] = 0
-
-            data[state_code][indicator_slug] += 1
-
-        with open(filename, 'wb') as f:
-            headers = ('State Code', 'State Name', 'Indicator', 'SMS Count')
-            excel_data = []
+            if indicator_slug not in district_level_data[state_code][district_code]:
+                district_level_data[state_code][district_code][indicator_slug] = 0
+
+            district_level_data[state_code][district_code][indicator_slug] += 1
+
+        for state_code, state_data in district_level_data.items():
```

Comment:
nit: it doesn't look like a second loop is necessary. Both `district_level_data` and `state_level_data` could be updated in the loop above.